cnd/src/main.rs

```diff
@@ -200,3 +203,14 @@ fn dump_config(settings: Settings) -> anyhow::Result<()> {
     println!("{}", serialized);
     Ok(())
 }
+
+fn lock_datadir(path: &PathBuf) -> anyhow::Result<File> {
+    let file = File::open(path)?;
+    match file.try_lock_exclusive() {
+        Err(e) => {
+            tracing::error!("Could not acquire lock on datadir: {:?}", path);
+            Err(anyhow::Error::from(e))
```

Comment:
Logging and passing on the error is a bit odd, you should be able to use `.context` here to add this information :)

https://docs.rs/anyhow/1.0.27/anyhow/trait.Context.html

Reply:
So basically doing this:

```rust
File::open(path)?.try_lock_exclusivly().with_context(|| format!("Could not acquire lock on data dir {}", path.display()))?
```

Reply:
Could also do an extension trait for `PathBuf` to put this code somewhere else and not having it inside `main.rs` :)

Reply:
> So basically doing this:
> 
> ```rust
> File::open(path)?.try_lock_exclusivly().with_context(|| format!("Could not acquire lock on data dir {}", path.display()))?
> ```

ok ill check his out, seems much cleaner